src/vm/arm64/asmhelpers.S

```diff
@@ -28,14 +28,12 @@ LEAF_ENTRY LazyMachStateCaptureState, _TEXT
 
     str lr, [x0, #LazyMachState_captureIp]
 
-    str fp, [x0, #LazyMachState_captureFp]
-
     // str instruction does not save sp register directly so move to temp register
     mov x1, sp
     str x1, [x0, #LazyMachState_captureSp]
 
     // save non-volatile registers that can contain object references
-    add x1, x0, #LazyMachState_captureX19_X28
+    add x1, x0, #LazyMachState_captureX19_X29
     stp x19, x20, [x1, #(16*0)]
     stp x21, x22, [x1, #(16*1)]
     stp x23, x24, [x1, #(16*2)]
```

Comment:
x29 should also be saved. This is missing . Take a look at same function in file src/vm/arm64/asmhelpers.asm